hathor/manager.py

```diff
@@ -778,9 +772,6 @@ def on_new_tx(self, tx: BaseTransaction, *, conn: Optional[HathorProtocol] = Non
 
         # if partial=False (the default) we don't even try to partially validate transactions
         if not partial or (metadata.validation.is_fully_connected() or tx.can_validate_full()):
-            if isinstance(tx, Transaction) and self.tx_storage.is_tx_needed(tx.hash):
-                tx._height_cache = self.tx_storage.needed_index_height(tx.hash)
```

Comment:
Why don't we need it anymore?

Reply:
The implementation is not using `_height_cache` anymore, so this isn't needed.

Reply:
Got it. Just to double check: Is it safe to remove it now? Or should we remove it together with the reward lock PR?